packages/reports/src/reports/editors/simple/SelectorUI.jsx

```diff
@@ -107,22 +164,82 @@ function SelectorUI(props) {
       onChange={setSelectorType}
       data={Object.values(SELECTOR_TYPE)}
     />
+    <Select
+      label="Options Type"
+      value={optionsType}
+      onChange={setOptionsType}
+      data={Object.values(OPTION_TYPE)}
+    />
+    {
+      optionsType === OPTION_TYPE.STATIC
+        ? <StaticSelectorOptionsEditor
+          staticOptions={options.static}
+          setStaticOptions={optionsState => setOptions({...options, static: optionsState})}
+        />
+        : <DynamicSelectorOptionsEditor
+          dynamicOptions={options.dynamic}
+          setDynamicOptions={optionsState => setOptions({...options, dynamic: optionsState})}
+          id={id}
+        />
+    }
+    {executeButton}
+  </>;
+}
+
+/** Component for defining and editing static, pre-defined options for a Selector block */
+function StaticSelectorOptionsEditor({staticOptions, setStaticOptions}) {
+
+  /** Add another option to selector */
+  const addOption = useCallback(
+    () => setStaticOptions([...staticOptions, {isDefault: !staticOptions.length, id: "", label: ""}]),
+    [staticOptions, setStaticOptions]
+  );
+
+  /** Change the values of an option at a specific index */
+  const editOption = useCallback(
+    (idx, newValues) => {
+      if (!newValues || idx >= staticOptions.length || idx < 0) return;
+
+      const newOptions = [...staticOptions];
+      newOptions.splice(idx, 1, {...staticOptions[idx], ...newValues});
+      setStaticOptions(newOptions);
+      // TODO - set error
```

Comment:
We have some great toasts for errors now - check out `import {useNotifications} from "@mantine/notifications";`